### ISSUE TITLE
OneSignal causing Resource Limit Exceeded (Wordpress Publish New Post)

### ISSUE THREAD

**Issue by idarek** [OneSignal/OneSignal-WordPress-Plugin#97]
Hi,
OneSignal causing an issue everytime we publish new material on our website. When we publish without notifying OneSignal, there is no issue.

Our hosting account throwing Resource Limit Exceeded and site is not available for approx 2-6 minutes each time. 

In my opinion, we don't have a lot subscribers (347), so the issue is strange.

**Comment by jasonpang** [comment]
Hi @idarek,

Can you please check whether your notification icon on your OneSignal dashboard is linking back to your site? Even if you're not sending notifications to many users, having all the notifications go to a couple hundred users at once will cause each subscriber's browser to download your notification icon separately and at the same time.

If your notification icon is causing the issue, we recommend hosting it on a CDN or image sharing service that can take the load more easily. If your notification icon isn't linking to your site, can you provide the log details on what the Resource Limit Exceeded is about?

**Comment by idarek** [comment]
Yes, Notification icon is linked to site. 
Yes, icon is hosted on same account.

How to know that notification icon is causing an issue?

I will try to place notification icon on different hosting to check next time when publishing something if you think that can cause an issue.

**Comment by jasonpang** [comment]
Hey @idarek,

The Resource Limit Exceeded errors are probably from the site under load serving a couple hundred requests all in the same second. You can check your server's logs to see more details about the Resource Limit Exceeded. Hosting the image elsewhere should fix the issue.

**Comment by idarek** [comment]
Yes, server showing heavy load and lot of requests. Don't have exact access to this logs, but that what got from hosting company. When OneSignal is off, everything is fine. Is Notification Icon such critical that can cause such issue?

ps. we using our hosting in cooperation with cloudflare.

**Comment by jasonpang** [comment]
Hey @idarek,

Each subscriber receiving a notification must download the icon from your server separately. The icon isn't sent with the notification. It's downloaded right before the notification is shown. Sending a notification to 350 subscribers can be like 350 visitors opening your site at the same time. 

Your server has to handle each request separately, and each request takes processing power and time. If your server isn't able to handle all the requests at the same time, any extra requests it can't handled will be dropped. This is probably what's causing your site to become unresponsive.

**Comment by idarek** [comment]
Thank you for help. Put an icon file to different hosting and thats solve an issue.

**Comment by idarek** [comment]
Hi,
I need to bring it back. I have moved hosting for an icon for 3 of websites that I manage.
One website got like 1000 subscribers. I have updated post and ticked to notify people about it and website just dies because of Resources Exceeded by multiple connection, which looks directly related to OneSignal.

Is there anything else that I can try to reduce this issue when notifying through OneSignal? #

**Comment by jasonpang** [comment]
@idarek,

We'll need more information about the Resource Limit Exceeded errors. Please contact your hosting provider to obtain the logs related to this issue and reach out to us at support@onesignal.com with the logs, thanks!

**Comment by idarek** [comment]
Hosting provide don't see anything unusual apart or extend contact with xmlrpc when issue exist.

ps. I have open #98 but you close it. #97 is marked as closed, where issue is still exist.

**Comment by YaronCarmi** [comment]
Hi All,
First, I would like to thanks you for this important Q&A, and especially to jasonpang, for the detailed explanation.
I also had this problem, which I resolved yesterday, after reading the explanation by jasonpang.
Here is what I did:

I had server crash for 20-30 seconds after each publish of 1800 subscribers.
I have a very powerfull SSD server, so it didn't look like a resource problem, but looks like some bug, caused by a wrong setup, done by me + wrong coding by OneSignal.

So I switched OFF the following button in the "OneSignal" control pannel in my Wordpress:
"Use the post's featured image for the notification icon"

By Doing so, the system is taking my favicon for the small image and keeps the post main image in the large image of the notification.

For some reason, this change resolved the bug.

I think that this bug must be fixed by OneSignal, regardless of the WorkAround I mentioned.

Good luck and thanks,
Yaron Carmi
